core/src/main/java/com/linecorp/armeria/server/logging/structured/ApacheThriftRpcServiceLogEntry.java

```diff
@@ -0,0 +1,98 @@
+import com.linecorp.armeria.common.thrift.ApacheThriftCall;
+import com.linecorp.armeria.common.thrift.ApacheThriftReply;
+import com.linecorp.armeria.common.thrift.ThriftCall;
+
+/**
+ * A representation and constructor of a service log which holds Apache Thrift based RPC invocation oriented
+ * information.
+ */
+public class ApacheThriftRpcServiceLogEntry extends CommonServiceLogEntry {
+    private final String thriftServiceName;
+    private final String thriftMethodName;
+    private final ApacheThriftCall thriftCall;
+    private final ApacheThriftReply thriftReply;
+
+    ApacheThriftRpcServiceLogEntry(long timestampMs,
+                                   long responseTimeMs,
+                                   String thriftServiceName,
+                                   String thriftMethodName,
+                                   ApacheThriftCall thriftCall,
+                                   ApacheThriftReply thriftReply) {
+        super(timestampMs, responseTimeMs);
+        this.thriftServiceName = thriftServiceName;
+        this.thriftMethodName = thriftMethodName;
+        this.thriftCall = thriftCall;
+        this.thriftReply = thriftReply;
+    }
+
+    /**
+     * Constructs {@link ApacheThriftRpcServiceLogEntry} from {@link RequestContext} and {@link ResponseLog}.
+     * Can be used as {@link ServiceLogEntryBuilder}.
+     */
+    public ApacheThriftRpcServiceLogEntry(RequestContext reqCtx, ResponseLog resLog) {
+        super(reqCtx, resLog);
+
+        RequestLog req = resLog.request();
+
+        RpcRequest rpcRequest = req.attr(RequestLog.RPC_REQUEST).get();
+        if (rpcRequest == null) {
+            // Request might be responded as error before reading arguments.
+            thriftServiceName = null;
+            thriftMethodName = null;
+            thriftCall = null;
+            thriftReply = null;
+            return;
+        }
+
+        if (!(rpcRequest instanceof ThriftCall)) {
+            throw new IllegalArgumentException(
+                    "expected ThriftCall instance for " + RequestLog.RPC_REQUEST + " but was " + rpcRequest);
+        }
+        ThriftCall thriftCall = (ThriftCall) rpcRequest;
+
+        thriftServiceName = thriftCall.serviceType().getCanonicalName()
+                                      .replaceAll("\\.Iface$", "")
+                                      .replaceAll("\\.AsyncIface$", "");
+        thriftMethodName = thriftCall.method();
+        this.thriftCall = (ApacheThriftCall) req.attr(RequestLog.RAW_RPC_REQUEST).get();
+        thriftReply = (ApacheThriftReply) resLog.attr(ResponseLog.RAW_RPC_RESPONSE).get();
+    }
+
+    public String getThriftServiceName() {
+        return thriftServiceName;
+    }
+
+    public String getThriftMethodName() {
+        return thriftMethodName;
+    }
+
+    public ApacheThriftCall getThriftCall() {
+        return thriftCall;
+    }
+
+    public ApacheThriftReply getThriftReply() {
```

Comment:
with breaking pojo convention we have to annotate each getter by `@JsonProperty`.
is it better than making these classes exceptional?